oneoffs/heatmap.py

```diff
@@ -0,0 +1,86 @@
+"""
+Used to plot a heatmap of the policy and value networks.
+Check FLAGS for default values for what models to load.
+
+Usage:
+python heatmap.py
+
+"""
+import sys; sys.path.insert(0, '.')
+
+import itertools
+import os
+
+import numpy as np
+import tensorflow as tf
+from tqdm import tqdm
+
+import go
+import oneoff_utils
+
+
+tf.app.flags.DEFINE_string(
+    "sgf_dir", "sgf/baduk_db/", "sgf database")
+
+tf.app.flags.DEFINE_string("model_dir", "saved_models",
+                           "Where the model files are saved")
+tf.app.flags.DEFINE_string("data_dir", "data/eval", "Where to save data")
+tf.app.flags.DEFINE_integer("idx_start", 150,
+                            "Only take models after given idx")
+tf.app.flags.DEFINE_integer("eval_every", 5,
+                            "Eval every k models to generate the curve")
+
+FLAGS = tf.app.flags.FLAGS
+
+
+def eval_for_policy(eval_positions, model_dir, data_dir, idx_start, eval_every):
+  model_paths = oneoff_utils.get_model_paths(model_dir)
+
+  print("Evaluating models {}-{}, eval_every={}".format(
+        idx_start, len(model_paths), eval_every))
+  for idx in tqdm(range(idx_start, len(model_paths), eval_every)):
+    if idx == idx_start:
+      player = oneoff_utils.load_player(model_paths[idx])
+    else:
+      oneoff_utils.restore_params(model_paths[idx], player)
+
+    pos_names, positions = zip(*eval_positions)
+    eval_probs, eval_values = player.network.run_many(positions)
+
+    for pos_name, probs, value in zip(pos_names, eval_probs, eval_values):
+      save_file = os.path.join(data_dir, "eval-{}-{}".format(pos_name, idx))
+
+      with open(save_file, "w") as data:
+        data.write("{},  {},  {}\n".format(
+            idx, value, ",".join(map(str, probs))))
+
+
+def positions_from_sgfs(sgf_files, include_empty=True):
+  # sgf_replay doesn't like empty sgf file
```

Comment:
i'm not super clear what these lines are doing; why would you want to 'include_empty' ever?  what's that for?

Reply:
Updated comment, I want to be able to analyse the empty board position (what does the policy heatmap look like when no stones are on the board) but sgf_replay considers this an error.